plugins/c8y_configuration_plugin/src/config.rs

```diff
@@ -1,90 +1,177 @@
+use crate::{error::ConfigManagementError, DEFAULT_PLUGIN_CONFIG_TYPE};
 use c8y_smartrest::topic::C8yTopic;
 use mqtt_channel::Message;
 use serde::Deserialize;
+use std::borrow::Borrow;
 use std::collections::HashSet;
 use std::fs;
 use std::hash::{Hash, Hasher};
-use std::path::PathBuf;
+use std::path::Path;
+use tedge_utils::file::FilePermissions;
 use tracing::{info, warn};
 
-#[derive(Deserialize, Debug, Eq, PartialEq, Default)]
+#[derive(Deserialize, Debug, Default)]
 #[serde(deny_unknown_fields)]
+struct RawPluginConfig {
```

Comment:
Why this name? Why the `Raw` prefix?

Reply:
The `RawPluginConfig` is used just to read and store the data from the config file. Afterwards, convert from `RawPluginConfig` to `PluginConfig`, while the convert, several exceptions will be checked.

I hope it's just naming issue. Do you have any suggestion?

Reply:
I'm okay for this `Raw` prefix, ... having no better suggestion ;-)